src/main/java/com/redhat/cloud/custompolicies/app/auth/RbacFilter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.cloud.custompolicies.app.auth;
+
+import com.redhat.cloud.custompolicies.app.RbacServer;
+import io.quarkus.cache.CacheResult;
+import java.io.IOException;
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+/**
+ * @author hrupp
+ */
+@Provider
+@Priority(Priorities.HEADER_DECORATOR +1)
+public class RbacFilter implements ContainerRequestFilter {
+
+
+  @Inject
+  @RestClient
+  RbacServer rbac;
+
+  @Inject
+  RhIdPrincipal user;
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    RbacRaw result;
+    try {
+      result = getRbacInfo(user.getRawRhIdHeader());
+    } catch (Throwable e) {
+      requestContext.abortWith(Response.status(Response.Status.FORBIDDEN).build());
+      return;
+    }
+
+    user.setRbac(result.canReadAll(),result.canWriteAll());
+    RhIdPrincipal userPrincipal = (RhIdPrincipal) requestContext.getSecurityContext().getUserPrincipal();
+    userPrincipal.setRbac(result.canReadAll(),result.canWriteAll());
```

Comment:
Why it adds it to `user` and then a `userPrincipal`? What's the difference between those two objects?

Reply:
The one is stored in the security context and the other made available via injection.
Currently the code only uses injection, but I make both available for convenience.
Btw. That was already in the previous implementation. It only was now pulled out into a separate class.